src/Provider/AbstractProvider.php

```diff
@@ -414,56 +409,75 @@ protected function sendRequest(RequestInterface $request)
      */
     public function getResponse(RequestInterface $request)
     {
-        $response = (string) $this->sendRequest($request)->getBody();
+        $response = $this->sendRequest($request);
         return $this->parseResponse($response);
     }
 
+
     /**
-     * Get the expected type of response for this provider.
+     * Parses response content as JSON.
      *
-     * @return string
+     * @param string $content JSON content from response body.
+     * @return array Decoded JSON data
+     * @throws UnexpectedValueException if the content could not be parsed.
      */
-    protected function getResponseType()
+    protected function parseJson($content)
     {
-        return static::RESPONSE_TYPE_JSON;
+        $content = json_decode($content, true);
+
+        if (json_last_error() !== JSON_ERROR_NONE) {
+            throw new UnexpectedValueException(sprintf(
+                "Failed to parse JSON response: %s",
+                json_last_error_msg()
+            ));
+        }
+
+        return $content;
     }
 
     /**
-     * Parse the response, according to the provider response type.
+     * Returns the content type header of a response.
+     *
+     * @param ResponseInterface $response
+     * @return string Semi-colon separated join of content-type headers.
+     */
+    protected function getContentType(ResponseInterface $response)
+    {
+        return join(";", (array) $response->getHeader('content-type'));
+    }
+
+    /**
+     * Parses the response, according to the response's content-type header.
      *
      * @throws UnexpectedValueException
-     * @param  string $response
+     * @param  ResponseInterface $response
      * @return array
      */
-    protected function parseResponse($response)
+    protected function parseResponse(ResponseInterface $response)
     {
-        $result = [];
-
-        switch ($this->getResponseType()) {
-            case static::RESPONSE_TYPE_JSON:
-                $result = json_decode($response, true);
-                if (JSON_ERROR_NONE !== json_last_error()) {
-                    throw new UnexpectedValueException('Unable to parse client response');
-                }
-                break;
-            case static::RESPONSE_TYPE_STRING:
-                parse_str($response, $result);
-                break;
+        $content = (string) $response->getBody();
+        $type = $this->getContentType($response);
+
+        if (strpos($type, "json") !== false) {
```

Comment:
Using `strpos` is more robust as a content-type header could be something like `test/json` or `application/json; charset=utf-8` which makes exact string matches more difficult.